torch/csrc/jit/constants.cpp

```diff
@@ -16,6 +16,32 @@ c10::OperatorOptions aliasAnalysisInternalSpecialCase() {
 }
 } // namespace
 
+bool insertableTensor(const at::Tensor& ten) {
+  return !ten.requires_grad();
+}
+
+bool insertableIValue(const IValue& ivalue) {
+  if (ivalue.isInt() || ivalue.isNone() || ivalue.isBool() ||
+      ivalue.isDouble() || ivalue.isString() || ivalue.isDevice()) {
+    return true;
+  }
+  if (ivalue.isTensor()) {
+    return insertableTensor(ivalue.toTensor());
+  }
+  if (ivalue.isList() || ivalue.isTuple()) {
```

Comment:
Does this mean that lists can be implicitly converted to tuple constants?

Reply:
I don't know what you mean